src/Ajax.php

```diff
@@ -301,24 +326,35 @@ public static function deleteShipment() {
         $orderId = $_POST['orderId'];
         $shipmentId = $_POST['id'];
 
+        $shipment = Helper::geShipmentWithJoinedItems($shipmentId);
         // Delete shipment record from the database.
         Helper::deleteShipment($shipmentId);
 
         // Order data.
         $order               = wc_get_order($orderId);
         $orderItems          = $order->get_items();
+
+        $order->add_order_note( sprintf( __( 'Shipment %s was deleted', 'trackmage' ), $shipment['tracking_number']), false, true );
+
         try {
             // Get HTML to return.
             ob_start();
             include TRACKMAGE_VIEWS_DIR . 'meta-boxes/order-shipments.php';
             $html = ob_get_clean();
+
+            ob_start();
+            $notes = wc_get_order_notes( array( 'order_id' => $orderId ) );
+            include WC()->plugin_path().'/includes/admin/meta-boxes/views/html-order-notes.php';
+            $notes_html = ob_get_clean();
+
         } catch (\Exception $e) {
             wp_send_json_error(['message' => $e->getMessage()]);
         }
 
         wp_send_json_success([
             'message' => __('Shipment deleted successfully!', 'trackmage'),
             'html' => $html,
+            'notes' => $notes_html
```

Comment:
Please move this block inside the catch body.
If an error happens inside the catch, it will not stop the execution and `html, notes_html` variables will not be initialized.

![image](https://user-images.githubusercontent.com/1675033/68879846-cd191400-0712-11ea-8abf-8f772c4f9568.png)